homeassistant/components/comfoair/fan.py

```diff
@@ -0,0 +1,116 @@
+"""Platform to control a Zehnder ComfoAir 350 ventilation unit."""
+import logging
+
+from homeassistant.components.fan import (
+    SPEED_HIGH,
+    SPEED_LOW,
+    SPEED_MEDIUM,
+    SPEED_OFF,
+    SUPPORT_SET_SPEED,
+    FanEntity,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import DOMAIN, SIGNAL_COMFOAIR_UPDATE_RECEIVED, ComfoAirModule
+
+_LOGGER = logging.getLogger(__name__)
+
+SPEED_MAPPING = {1: SPEED_OFF, 2: SPEED_LOW, 3: SPEED_MEDIUM, 4: SPEED_HIGH}
+
+
+async def async_setup_platform(hass, conf, async_add_entities, discovery_info):
+    """Set up the ComfoAir fan platform."""
+    unit = hass.data[DOMAIN]
+
+    async_add_entities([ComfoAirFan(hass, ca=unit)], True)
+
+
+class ComfoAirFan(FanEntity):
+    """Representation of the ComfoAir fan platform."""
+
+    def __init__(self, hass, ca: ComfoAirModule) -> None:
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.